This patch fixes a bug in version 1.37.24 when the wrong tape is initially loaded, the
Storage daemon thinks that the desired tape is already loaded.

Apply the patch to 1.37.24 with:

  cd <bacula-source>
  patch -p0 <1.37.24-wrong-tape.patch
  make
  ...

Index: src/version.h
===================================================================
RCS file: /cvsroot/bacula/bacula/src/version.h,v
retrieving revision 1.498
diff -u -r1.498 version.h
--- src/version.h	18 Jun 2005 13:37:19 -0000	1.498
+++ src/version.h	20 Jun 2005 08:09:56 -0000
@@ -1,8 +1,8 @@
 /* */
 #undef  VERSION
-#define VERSION "1.37.24"
-#define BDATE   "18 June 2005"
-#define LSMDATE "18Jun05"
+#define VERSION "1.37.25"
+#define BDATE   "20 June 2005"
+#define LSMDATE "20Jun05"
 
 /* Debug flags */
 #undef  DEBUG
Index: src/stored/mount.c
===================================================================
RCS file: /cvsroot/bacula/bacula/src/stored/mount.c,v
retrieving revision 1.88
diff -u -r1.88 mount.c
--- src/stored/mount.c	18 Jun 2005 13:22:36 -0000	1.88
+++ src/stored/mount.c	20 Jun 2005 08:09:56 -0000
@@ -190,7 +190,7 @@
       return false;
    }
 
-   Dmsg2(100, "dirVol=%s dirStat=%s\n", dcr->VolumeName,
+   Dmsg2(100, "Want dirVol=%s dirStat=%s\n", dcr->VolumeName,
       dcr->VolCatInfo.VolCatStatus);
    /*
     * At this point, dev->VolCatInfo has what is in the drive, if anything,
@@ -230,6 +230,7 @@
       memcpy(&VolCatInfo, &dcr->VolCatInfo, sizeof(VolCatInfo));
       memcpy(&devVolCatInfo, &dev->VolCatInfo, sizeof(devVolCatInfo));
       /* Check if this is a valid Volume in the pool */
+      bstrncpy(dcr->VolumeName, dev->VolHdr.VolumeName, sizeof(dcr->VolumeName));
       if (!dir_get_volume_info(dcr, GET_VOL_INFO_FOR_WRITE)) {
          /* Restore desired volume name, note device info out of sync */
          /* This gets the info regardless of the Pool */
